starknet/src/fact_registry/contract.cairo

```
@@ -111,15 +111,16 @@ pub mod FactRegistry {
             block: u64,
             proof_sizes_bytes: Array<usize>,
             proofs_concat: Array<u64>,
-        ) {
+        ) -> Result<bool, felt252> {
             let state_root = self.l1_headers_store.read().get_state_root(block);
             assert!(state_root != 0, "FactRegistry: block state root not found");
             let proof = self
                 .reconstruct_ints_sequence_list(proofs_concat.span(), proof_sizes_bytes.span());
             let result = verify_proof(account.to_words64(), state_root.to_words64(), proof.span());
             match result {
-                Option::None => { panic!("FactRegistry: account not found"); },
-                Option::Some(result) => {
+                Result::Err => { Result::Err('FactRegistry: account not found') },
```

Here we can return the error propagated by `verify_proof`